Automattic/newspack-plugin
# fix: only enqueue salesforce admin JS when appropriate

### All Submissions:






### Changes proposed in this Pull Request:



Tightens the conditions for when Newspack's Salesforce integration admin JS is enqueued. It should only be loaded when viewing a single order admin page, and when the site has an active Salesforce connection.

### How to test the changes in this Pull Request:

1. On a test site, set your Reader Revenue platform to "Newspack" and establish a Salesforce connection in **Newspack > Reader Revenue > Salesforce**. Ping me for test credentials if needed.
2. On `master`, go to **WooCommerce > Orders** and observe a JS console error:

```
Uncaught TypeError: statusMarker is null
```

Note that this error may also prevent the Orders page from being scrollable.

3. Disconnect Salesforce in the Reader Revenue wizard (or set your Reader Revenue platform to something other than Newspack), then edit a single order in **WooCommerce > Orders**. Observe that the "Not synced" flag is still shown.
4. Reconnect Salesforce, then check out this branch and repeat steps 2-3. Confirm that a.) the error is cleared in the Orders list admin page, and that the page is scrollable; and b.) that the "Not synced" flag is not shown for single orders if there's no active Salesforce connection.

### Other information:
